src/crds_gossip_pull.rs

```diff
@@ -222,8 +222,8 @@ mod test {
         // This is because the heaviest nodes will have very similar weights
         let min_balance = E.powf(3000_f32.ln() - 0.5);
         let now = 1024;
-        // try upto 10 times because of rng
-        for _ in 0..10 {
+        // try up to 20 times because of rng
```

Comment:
sigh, yes can't really help this one. I tried making it as biased as possible but there's _always_ a chance. Maybe we should just ignore this and force me to rewrite something. :S

Reply:
Sagar, can you take this one and land it by itself?

Reply:
@sagar-solana - I did 10ad536e0955c182f6bd2fdcfdbd91dd9833dfc4 for now since I keep seeing this test fail in CI

Reply:
@mvines check out https://github.com/solana-labs/solana/pull/2954, I got rid of the rng in the test.

Reply:
\o/ d1e1258f97462516c266bd73a287a451032881db